src/topology/discrete_quotient.lean

```diff
@@ -198,6 +200,23 @@ lemma of_le_proj_apply {A B : discrete_quotient X} (h : A ≤ B) (x : X) :
 
 end of_le
 
+/--
+When X is discrete, there is a `semilattice_inf_bot` instance on `discrete_quotient X`
+-/
+instance [discrete_topology X] : semilattice_inf_bot (discrete_quotient X) :=
+{ bot :=
+  { rel := λ a b , a = b,
+    equiv := ⟨λ a, rfl, λ a b h, h.symm, λ a b c h1 h2, h1.trans h2⟩,
+    clopen := λ x, ⟨is_open_discrete _, is_closed_discrete _⟩ },
+  bot_le := by { rintro S a b (h : a = b), rw h, exact S.refl _ },
+  ..(infer_instance : semilattice_inf _) }
```

Comment:
```suggestion
  ..(by apply_instance : semilattice_inf (discrete_quotient X)) }
```

This should solve the issue from Eric's comment.

Reply:
I don't really mind about this much since it now adds verbosity.